api-examples/pretrain-transformer-lm.py

```diff
@@ -579,11 +589,28 @@ def train():
         for i, batch in enumerate(train_loader):
             x, y = batch
             inputs = {'x': x.to(args.device)}
-            labels = y.to(args.device).transpose(0, 1).contiguous()
+            labels = y.to(args.device)
+            if args.mlm:
+                # Replace 15% of tokens
+                masked_indices = torch.bernoulli(torch.full(labels.shape, 0.15)).byte()
+                # Anything not masked is 0 so no loss
+                labels[~masked_indices] = 0
+                # Of the masked items, mask 80% of them with [MASK]
+                indices_replaced = torch.bernoulli(torch.full(labels.shape, 0.8)).byte() & masked_indices
+                inputs[indices_replaced] = mask_value
+                # Replace 10% of them with random words, rest preserved for auto-encoding
+                indices_random = torch.bernoulli(torch.full(labels.shape, 0.5)).byte() & masked_indices & ~indices_replaced
+                random_words = torch.randint(args.d_model, labels.shape, dtype=torch.long, device=args.device)
```

Comment:
Are there tokens that we shouldn't include in the random words (Possible by setting the low value in the random? Things like `<PAD>` and `<GO>`?

Reply:
Possibly, this is a good point.  Also we know that this BPE-based masking is sub-optimal.  For now would just like to get an initial attempt in there.